azure-iot-device/samples/advanced-hub-scenarios/register_x509.py

```diff
@@ -0,0 +1,51 @@
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+# --------------------------------------------------------------------------
+
+
+# This is for illustration purposes only. The sample will not work currently.
+
+import os
+import logging
+import asyncio
+from azure.iot.device import X509SecurityClient, X509
+from azure.iot.device.aio import X509ProvisioningDeviceClient
+
+
+logging.basicConfig(level=logging.INFO)
+
+provisioning_host = os.getenv("PROVISIONING_HOST")
+id_scope = os.getenv("PROVISIONING_IDSCOPE")
+registration_id = os.getenv("DPS_X509_REGISTRATION_ID")
+
+
+async def main():
+    async def register_device():
+        x509_cert = X509(
+            cert_file="../../cert/device_x509.pem", key_file="../../cert/device_x509_key.pem"
```

Comment:
>"../../cert/device_x509.pem" [](start = 22, length = 28)

nit: i would make these env variables as well. i'm not sure we should suggest customers shipping with hardcoded file paths.